### PR TITLE
KEP-596: Move CSIInlineVolume feature to GA

### PR DESCRIPTION
**Pull request by dobsonj**
#### What type of PR is this?

/kind feature
/sig storage
/milestone v1.25
/triage accepted
/priority important-soon

#### What this PR does / why we need it:

This updates the feature gates to move CSIInlineVolume from beta to GA in 1.25.

#### Which issue(s) this PR fixes:

Enhancement: [KEP-596](https://github.com/kubernetes/enhancements/issues/596).

#### Special notes for your reviewer:

/cc @gnufied @jsafrane @xing-yang @jingxu97

I'd like to start review and e2e tests, but
/hold for remaining [graduation criteria](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/596-csi-inline-volumes/README.md#ga) before merging.

The API changes are only for removal of the feature gate and references to it in the generated docs.
/kind api-change
/label api-review

#### Does this PR introduce a user-facing change?

```release-note
The CSIInlineVolume feature has moved from beta to GA.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

- KEP: https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/596-csi-inline-volumes/README.md
- kubernetes/website: https://kubernetes.io/docs/concepts/storage/ephemeral-volumes
- kubernetes-csi/docs: https://kubernetes-csi.github.io/docs/ephemeral-local-volumes.html

### CODE REVIEW

**Comment by dobsonj** [comment]
```
[sig-network] Services should fail health check node port if there are only terminating endpoints [Feature:ProxyTerminatingEndpoints] 
```
/retest

**Comment by liggitt** [comment]
cc @msau42 for early review, putting in the api-review backlog until graduation criteria are met

File: pkg/volume/csi/csi_test.go [hunk 1]
```diff
@@ -244,8 +220,6 @@ func TestCSI_VolumeAll(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIInlineVolume, !test.disableFSGroupPolicyFeatureGate)()
```
Comment (msau42) [inline pkg/volume/csi/csi_test.go:247]: do you know why CSIInlineVolume was tied to fsgroup policy feature gate?

**Review by msau42 — COMMENTED** [review]
api changes lgtm

File: pkg/volume/csi/csi_test.go (hunk 1 above)
Comment (dobsonj) [reply pkg/volume/csi/csi_test.go:247]: There was just the one test case [added here](https://github.com/kubernetes/kubernetes/commit/4d2d063635c86aceb626d88c1d785e0463109df8#diff-805a29cf10e2aafaae4e1665d310fa4edd041e32da71bba9cf06f7200368cd70R96-R114) to test `supportsFSGroup` with the feature gate disabled because of this check:
https://github.com/kubernetes/kubernetes/blob/96a32be18dfbd38ff5cb1df4d68528e2cf226958/pkg/volume/csi/csi_mounter.go#L417-L421
In other words, the current behavior (without my PR) is to fail setting FSGroup on CSI Inline Volumes when the feature gate is disabled. TBH, I don't think we could actually get that far if the feature gate is disabled since the volume couldn't be created in the first place.

Comment (dobsonj) [reply pkg/volume/csi/csi_test.go:247]: Ah, and to be clear I think `disableFSGroupPolicyFeatureGate` was a misleading variable name since it only ever controlled the `CSIInlineVolume` feature gate.

**Comment by liggitt** [comment]
API changes lgtm as well.
once the functional part of this has lgtm, readd me and I'll tag for API approval

**Comment by dobsonj** [comment]
/unhold
Graduation criteria updated in https://github.com/kubernetes/enhancements/pull/3445
/assign @msau42 
for lgtm / functional review

/retest
The only test failure is this one due to time out, but it's not a required test:
```
[sig-network] Services should fail health check node port if there are only terminating endpoints [Feature:ProxyTerminatingEndpoints] 
```

**Comment by cici37** [comment]
Hello 👋, 1.25 Release Lead here.

Unfortunately, this enhancement did not meet the [code freeze](https://github.com/kubernetes/sig-release/blob/master/releases/release_phases.md#code-freeze) criteria because there are still unmerged k/k code PRs.

If you still wish to progress this enhancement in v1.25, please file an [exception request](https://github.com/kubernetes/sig-release/blob/master/releases/EXCEPTIONS.md). Thank you so much!

/milestone clear

**Comment by dobsonj** [comment]
Last push rebases to resolve a merge conflict, adds e2e tests to be promoted to conformance, and [removes one more reference](https://github.com/kubernetes/kubernetes/pull/111258/commits/965446c7265354161b9013806eca8dad3a74af0e#diff-19b27e79a2dcb337f58b0f7f3477a29d29498b5d135fbb67e9f858cf1dae222cL157-R157) to this as a beta feature.
Latest test failures appear unrelated:
```
Kubernetes e2e suite: [It] [sig-node] Probing container should mark readiness on pods to false while pod is in progress of terminating when a pod has a readiness probe expand_more

k8s.io/kubernetes/test/integration/daemonset: TestDSCUpdatesPodLabelAfterDedupCurHistories 

Kubernetes e2e suite: [It] [sig-network] Services should fail health check node port if there are only terminating endpoints [Feature:ProxyTerminatingEndpoints] expand_more
```
/retest

**Comment by dobsonj** [comment]
Last test run was clean, but had to rebase again for conflicts in strategy.go and strategy_test.go.

**Comment by msau42** [comment]
/lgtm
/approve

Thanks for adding the new test!

**Comment by dobsonj** [comment]
> API changes lgtm as well. once the functional part of this has lgtm, readd me and I'll tag for API approval

/assign @liggitt 
for API approval

**Comment by liggitt** [comment]
/approve
for API change

**Comment by cici37** [comment]
Hello 👋, 1.25 Release Lead here.

The [exception request](https://groups.google.com/g/kubernetes-sig-release/c/3kmMHPHXrgk/m/md41QANHAQAJ) is approved and your updated deadline to make any changes to your PR is 9:00 AM PST Monday 8th August 2022. Thank you!

/milestone v1.25
